acnportal/acnsim/simulator.py

```diff
@@ -38,8 +40,8 @@ def __init__(self, network, scheduler, events, start, period=1, max_recomp=None,
         self.verbose = verbose
 
         # Information storage
-        self.pilot_signals = {station_id: [] for station_id in self.network.space_ids}
-        self.charging_rates = {station_id: [] for station_id in self.network.space_ids}
+        self.pilot_signals = np.zeros((len(self.network.station_ids), len(self.event_queue)))
```

Comment:
Should not use the length of the event queue, but rather the timestamp of the last event in the queue.